### PR TITLE
Fix: Removing exclusion patterns in 'Highly recommended' widget which are …

### LINKED ISSUE #1620: "Highly recommended" exclusion patterns in "Manage profiles" GUI dialog should show only missing patterns
## Current status

`rsync` and therefor BiT do allow to specify "exclude file patterns" for files that shall not be `rsync`ed:

Currently the GUI always shows all recommended (= default) exclusion file patterns:

image

## Feature request

It would be more helpful to show only those default exclusions patterns as "highly recommended" that are not already contained in the actual exclusion list.

## Relevant code

https://github.com/bit-team/backintime/blob/d583cf8a4b7de4271f897f07a94d0fbe07f9caff/qt/settingsdialog.py#L556-L561

https://github.com/bit-team/backintime/blob/d583cf8a4b7de4271f897f07a94d0fbe07f9caff/common/config.py#L98-L119

## Related to

- #1583
- #1618
- #1594
- #1382
- #1215
- #972

### PR DESCRIPTION
**Pull request by Prakul95**
…already excluded in manage profile

created a set for excluded list items and created a new list of recommended list which are not present in excluded

Would fix  #1620

### CODE REVIEW

**Comment by buhtz** [comment]
Hello Prakul,
welcome to the project. Thanks for this contribution and investing your time in the project.

Did you choose this task because of the GoodFirstIssue label?

Have you tested your solution? Have you started Back In Time?

I do got an error:

```
  File "/home/user/mluCloud/my.work/Prakul95.backintime/qt/settingsdialog.py", line 559, in __init__
    setExclude = set(self.listExclude)
                 ^^^^^^^^^^^^^^^^^^^^^
TypeError: 'QTreeWidget' object is not iterable
```

**Comment by aryoda** [comment]
@Prakul95 Thanks a lot for your contribution!

Could you please also add an entry into the  [`CHANGES`](https://github.com/bit-team/backintime/blob/dev/CHANGES) file of your clone which will be part of the release notes in the next release then... THX :smile:

File: qt/settingsdialog.py [hunk 1]
```diff
@@ -556,7 +556,10 @@ def __init__(self, parent):
         label = QLabel(_('Highly recommended') + ':', self)
         qttools.setFontBold(label)
         layout.addWidget(label)
-        label = QLabel(', '.join(sorted(self.config.DEFAULT_EXCLUDE)), self)
+        setExclude = set(self.listExclude)
+        label = QLabel(', '.join(sorted([item for item in self.config.DEFAULT_EXCLUDE if item not in setExclude ])), self)
+        label.setWordWrap(True)
```
Comment (aryoda) [inline qt/settingsdialog.py:561]: Looks like the same LOC as one line later... Duplicated code?

**Comment by buhtz** [comment]
Dear @Prakul95 ,
are you still with us?

**Comment by buhtz** [comment]
Dear @Prakul95 ,
my intention is to learn and to improve the handling of new and beginner contributors.

So please let me know if we can improve something in this process. I am interested in the reasons why you don't work on this PR anymore? Maybe you wasn't aware that a contribution do not stop with opening a PR but nearly every PR need some extra work?

Kind
Christian

**Comment by buhtz** [comment]
Close and replaced by #1665

**Pull request closed without merging**